_posts/2023-12-20-in-progress-wildfly-feature-on-openshift.adoc

```diff
@@ -0,0 +1,312 @@
+```
+SIMPLE_WEBAPP_OIDC_URL=https://$(oc get route simple-webapp-oidc --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Application URL:              $SIMPLE_WEBAPP_OIDC_URL/simple-webapp-oidc" &&
+echo "Valid redirect URI:           $SIMPLE_WEBAPP_OIDC_URL/simple-webapp-oidc/*" &&
+echo ""
+```
+
+== Finish Configuring RH-SSO 
+Copy the link for your webapp and go back to the admin console for RH-SSO and add `<Simple-webapp-oidc-url>/simple-webapp-oidc/*`
+to the `Valid Redirect URIs` field. Notice that the link starts with `https`. We will be changing it to `http` since we
+have not performed any ssl configuration.
+
+== Access the Application
+Now, let’s try accessing our application using <Simple-webapp-oidc-url>/simple-webapp-oidc. 
+
+Click on "Access Secured Servlet".
+
+Now, you’ll be redirected to RH-SSO's login page. If you click on the url on the search bar, you will see the scope values specified in the `redirect-uri` field with the different scope values separated by a `+`. You will also notice that a new scope value: `openid`. This indicates that we are going to be using OpenID Connect to authenticate the user. 
+
+Log in with `Alice` and the password that you set when configuring RH-SSO.
+
+Next, you’ll be redirected back to our application, and you should see the "Secured Servlet" page. That means that we were able to successfully log in to our application using the RH-SSO OpenID provider!
+
+This page will display the current principal, and a list of claim values obtained using the scope values you configured. This is what it will look like: 
+```
+Secured Servlet
+
+Current Principal 'alice'
+Claims received using additional scope values:
+Using the "profile" scope, we got User's name: Alice Smith
+Using the "email" scope, we got email verified: true
+Using the "microprofile-jwt" scope, we got the user's groups: [default-roles-myrealm, offline_access, admin, uma_authorization, user]
+```
+
+Note that the value for Current Principal may be different, since that is the client secret. 
+
+Notice that there are no claims obtained using the `offline_access` scope. To learn more about what this scope value does, please refer to the https://openid.net/specs/openid-connect-core-1_0.html#OfflineAccess[ OpenID Documentation]. 
+
+== Summary 
+This post demonstrated how to deploy an OIDC secured WildFly application on OpenShift using Developer Sandbox. We also looked into how to configure additional scope values to request specific claim values. For more details on the `elytron-oidc-client` subsystem, please check out the https://docs.wildfly.org/30/Admin_Guide.html#Elytron_OIDC_Client[documentation]. Details on the `scope` attribute will be available once the feature is released. For more details on WildFly on OpenShift, you can check out https://cloud.redhat.com/blog/getting-started-with-wildfly[this guide].
+
```

Comment:
You could also mention something like:

If you have any questions or comments about this new upcoming feature, please let us know over on the [WildFly User Forum](https://groups.google.com/g/wildfly).

Reply:
fixed!